proposals/0030-core-image-story.md

```diff
@@ -0,0 +1,427 @@
+
+In tandem with the `src/assets` folder, we'd like to introduce a way for users to specify that a specific property needs to refer to a valid asset from the `src/assets` folder:
+
+```ts
+import { asset, defineCollection, z } from "astro:content";
+
+const blogCollection = defineCollection({
+  schema: z.object({
+    title: z.string(),
+    image: asset({ width: 1120, height: 1120 }),
+  }),
+});
+```
+
+Additionally, assets referred this way will be transformed automatically to the same shape as if the image was imported (see section below.), so they're ready to be used optimally. However, optimization / resizing is left to the user to do using the JavaScript API if they desire to.
+
+## New ESM shape for images imports
+
+Currently, importing images in Astro returns a simple `string` with the path of the image. The `@astrojs/image` integration and this RFC enhance this by instead returning the following shape: `{src: string, width: number, height: number, format: string}`.
+
+This shape has multiple benefits:
+
+- It allows users to easily construct `img` tags with no CLS, without needing to opt-in the entire `Image` pipeline if they don't require it:
+
+```astro
+---
+import image from "../my_image.png";
+---
+
+<img src={image.src} width={image.width} height={image.height} />
+```
+
+- This shapes gives all the information needed to build your own image integration. Depending on the transformer used, you'll often need the original metadata (ex: width and height) to resize the image while keeping the aspect ratio. Or, you might need a different encoder depending on the file format. With this output, all of this is given without needing further steps.
+
+## Image component
+
+A component that outputs a simple `<img> `tag with all the required attributes needed to show the image.
+
+### Properties
+
+#### width and height
+
+Those properties are used for resizing the image and ensuring there's no CLS. In most cases, the `width` and `height` would be automatically inferred from the source file and not passing them would just result in the same dimensions.
+
+The only time those properties are required is for remote images.
+
+#### format
+
+`format` is used to set the resulting wanted format for the image.
+
+Default is left to the services to choose. But for the services exposed by Astro, this property is always optional and is set to `webp`.
+
+#### quality
+
+`quality` inform the quality to use when transforming the image. It can either be a number from `0` to `100`, or a preset from `low`, `medium`, `high`, `max`. Specific implementation here is left to the image service used, as they do not necessarily all abide by the same rules.
+
+Default is left to the services to choose. But for the services exposed by Astro, this property is always optional.
+
+#### alt
+
+`alt` is a required property in all cases when using the component. It can be explicitly set to `""`, for cases where an alt-text is not required.
+
```

Comment:
Are there cases where `alt` is not needed?? 😱

Reply:
From MDN:

> Setting this attribute to an empty string (alt="") indicates that this image is not a key part of the content (it's decoration or a tracking pixel), and that non-visual browsers may omit it from [rendering](https://developer.mozilla.org/en-US/docs/Glossary/Rendering_engine).